File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> myMap;
        vector<int> result;
        
        for(int i = 0; i < nums.size(); i++) {
            int comp = target - nums[i];
            
            if(myMap.find(comp) != myMap.end()) {
                result.push_back(myMap[comp]);
                result.push_back(i);
            }
            
            myMap[nums[i]] = i;
        }
        
        return result;
    }
};

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mymap = {}
        for i, comp in enumerate(nums):
            if target - comp in mymap:
                return [mymap[target - comp], i]
            else:    
                mymap[comp] = i
        return []
          